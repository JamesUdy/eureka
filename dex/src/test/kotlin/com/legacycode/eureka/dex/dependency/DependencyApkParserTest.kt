package com.legacycode.eureka.dex.dependency

import com.legacycode.eureka.dex.test.TestApk
import com.legacycode.eureka.dex.test.TestGraphBuilder
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

class DependencyApkParserTest {
  @Test
  fun name() {
    // given
    val testApk = TestApk("wikipedia.apk")
    val parser = DependencyApkParser(testApk.file)

    // when
    val adjacencyList = parser.buildDependencyGraph()

    // then
    val dependencyGraph = adjacencyList.graph(TestGraphBuilder())
    Approvals.verify(dependencyGraph)
  }
}
