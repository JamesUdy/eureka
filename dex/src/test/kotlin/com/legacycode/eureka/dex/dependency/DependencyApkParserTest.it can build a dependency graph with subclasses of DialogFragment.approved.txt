Lorg/joinmastodon/android/fragments/ComposeFragment; -> Lorg/joinmastodon/android/ExternalShareActivity;
Lorg/joinmastodon/android/fragments/ComposeFragment; -> Lorg/joinmastodon/android/MainActivity;
Lorg/joinmastodon/android/fragments/ThreadFragment; -> Lorg/joinmastodon/android/MainActivity;
Lorg/joinmastodon/android/fragments/ProfileFragment; -> Lorg/joinmastodon/android/MainActivity;
Lorg/joinmastodon/android/fragments/SplashFragment; -> Lorg/joinmastodon/android/MainActivity;
Lorg/joinmastodon/android/fragments/HomeFragment; -> Lorg/joinmastodon/android/MainActivity;
Lorg/joinmastodon/android/fragments/onboarding/AccountActivationFragment; -> Lorg/joinmastodon/android/MainActivity;
Lorg/joinmastodon/android/MainActivity; -> Lorg/joinmastodon/android/OAuthActivity;
Lorg/joinmastodon/android/fragments/discover/DiscoverAccountsFragment; -> Lorg/joinmastodon/android/fragments/discover/DiscoverFragment;
Lorg/joinmastodon/android/fragments/discover/TrendingHashtagsFragment; -> Lorg/joinmastodon/android/fragments/discover/DiscoverFragment;
Lorg/joinmastodon/android/fragments/discover/LocalTimelineFragment; -> Lorg/joinmastodon/android/fragments/discover/DiscoverFragment;
Lorg/joinmastodon/android/fragments/discover/DiscoverNewsFragment; -> Lorg/joinmastodon/android/fragments/discover/DiscoverFragment;
Lorg/joinmastodon/android/fragments/discover/DiscoverPostsFragment; -> Lorg/joinmastodon/android/fragments/discover/DiscoverFragment;
Lorg/joinmastodon/android/fragments/discover/SearchFragment; -> Lorg/joinmastodon/android/fragments/discover/DiscoverFragment;
Lorg/joinmastodon/android/fragments/ThreadFragment; -> Lorg/joinmastodon/android/fragments/discover/SearchFragment;
Lorg/joinmastodon/android/fragments/ProfileFragment; -> Lorg/joinmastodon/android/fragments/discover/SearchFragment;
Lorg/joinmastodon/android/fragments/report/ReportCommentFragment; -> Lorg/joinmastodon/android/fragments/report/ReportAddPostsChoiceFragment;
Lorg/joinmastodon/android/fragments/report/ReportRuleChoiceFragment; -> Lorg/joinmastodon/android/fragments/report/ReportReasonChoiceFragment;
Lorg/joinmastodon/android/fragments/report/ReportAddPostsChoiceFragment; -> Lorg/joinmastodon/android/fragments/report/ReportReasonChoiceFragment;
Lorg/joinmastodon/android/fragments/report/ReportDoneFragment; -> Lorg/joinmastodon/android/fragments/report/ReportReasonChoiceFragment;
Lorg/joinmastodon/android/fragments/report/ReportAddPostsChoiceFragment; -> Lorg/joinmastodon/android/fragments/report/ReportRuleChoiceFragment;
Lorg/joinmastodon/android/fragments/SettingsFragment; -> Lorg/joinmastodon/android/fragments/onboarding/AccountActivationFragment;
Lorg/joinmastodon/android/fragments/onboarding/OnboardingFollowSuggestionsFragment; -> Lorg/joinmastodon/android/fragments/onboarding/AccountActivationFragment;
Lorg/joinmastodon/android/MainActivity; -> Lorg/joinmastodon/android/fragments/onboarding/AccountActivationFragment;
Lorg/joinmastodon/android/fragments/onboarding/SignupFragment; -> Lorg/joinmastodon/android/fragments/onboarding/GoogleMadeMeAddThisFragment;
Lorg/joinmastodon/android/fragments/onboarding/GoogleMadeMeAddThisFragment; -> Lorg/joinmastodon/android/fragments/onboarding/InstanceRulesFragment;
Lorg/joinmastodon/android/fragments/onboarding/OnboardingProfileSetupFragment; -> Lorg/joinmastodon/android/fragments/onboarding/OnboardingFollowSuggestionsFragment;
Lorg/joinmastodon/android/fragments/onboarding/InstanceRulesFragment; -> Lorg/joinmastodon/android/fragments/onboarding/InstanceCatalogSignupFragment;
Lorg/joinmastodon/android/fragments/HomeTimelineFragment; -> Lorg/joinmastodon/android/fragments/HomeFragment;
Lorg/joinmastodon/android/fragments/NotificationsFragment; -> Lorg/joinmastodon/android/fragments/HomeFragment;
Lorg/joinmastodon/android/fragments/ProfileFragment; -> Lorg/joinmastodon/android/fragments/HomeFragment;
Lorg/joinmastodon/android/fragments/discover/DiscoverFragment; -> Lorg/joinmastodon/android/fragments/HomeFragment;
Lorg/joinmastodon/android/fragments/onboarding/OnboardingFollowSuggestionsFragment; -> Lorg/joinmastodon/android/fragments/HomeFragment;
Lorg/joinmastodon/android/fragments/onboarding/InstanceCatalogSignupFragment; -> Lorg/joinmastodon/android/fragments/SplashFragment;
Lorg/joinmastodon/android/fragments/onboarding/InstanceChooserLoginFragment; -> Lorg/joinmastodon/android/fragments/SplashFragment;
Lorg/joinmastodon/android/fragments/ComposeImageDescriptionFragment; -> Lorg/joinmastodon/android/fragments/ComposeFragment;
Lorg/joinmastodon/android/fragments/NotificationsListFragment; -> Lorg/joinmastodon/android/fragments/NotificationsFragment;
Lorg/joinmastodon/android/fragments/ProfileAboutFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/AccountTimelineFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/ComposeFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/account_list/FollowerListFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/account_list/FollowingListFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/report/ReportReasonChoiceFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/BookmarkedStatusListFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/FavoritedStatusListFragment; -> Lorg/joinmastodon/android/fragments/ProfileFragment;
Lorg/joinmastodon/android/fragments/onboarding/AccountActivationFragment; -> Lorg/joinmastodon/android/fragments/SettingsFragment;
Lorg/joinmastodon/android/MainActivity; -> Lorg/joinmastodon/android/fragments/SettingsFragment;
Lorg/joinmastodon/android/fragments/ThreadFragment; -> Lorg/joinmastodon/android/fragments/NotificationsListFragment;
Lorg/joinmastodon/android/fragments/ProfileFragment; -> Lorg/joinmastodon/android/fragments/NotificationsListFragment;
Lorg/joinmastodon/android/fragments/ThreadFragment; -> Lorg/joinmastodon/android/fragments/StatusListFragment;
Lorg/joinmastodon/android/fragments/ComposeFragment; -> Lorg/joinmastodon/android/fragments/HashtagTimelineFragment;
Lorg/joinmastodon/android/fragments/ComposeFragment; -> Lorg/joinmastodon/android/fragments/HomeTimelineFragment;
Lorg/joinmastodon/android/fragments/SettingsFragment; -> Lorg/joinmastodon/android/fragments/HomeTimelineFragment;
